src/websocket/pushmessage.cpp

```diff
@@ -76,15 +77,27 @@ void PushMessage::parseMessage(const QString& message) {
 PushMessage::MessageType PushMessage::messageTypeFromString(
     const QString& str) {
   if (str == "DEVICE_DELETED") return MessageType_DeviceDeleted;
-
 #ifdef UNIT_TEST
   if (str == "TEST_MESSAGE") return MessageType_TestMessage;
 #endif
-
   logger.debug() << "Unknown message type" << str;
   return MessageType_UnknownMessage;
 }
 
+// static
+const QString PushMessage::messageTypeToString(MessageType& type) {
+  switch (type) {
+    case MessageType_DeviceDeleted:
+      return "DEVICE_DELETED";
```

Comment:
what about if we use QVariant to do this? If MessageType is a Q_ENUM, we can do: ` QVariant::fromValue(type).toString()`

Reply:
Ah I think that is the issue, MessateType is not yet a Q_ENUM!